### PR TITLE
Add linux os release info to primary key

### PR DESCRIPTION
**Pull request by panticmilos**
**Description:**
This PR provides expanded cache primary and restore keys for Linux distributions

**Related issue:**
#432

**Check list:**
- [x] Mark if documentation changes are required.
- [ ] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

File: src/cache-distributions/pip-cache.ts [hunk 1]
```diff
@@ -57,8 +57,19 @@ class PipCache extends CacheDistributor {
 
   protected async computeKeys() {
     const hash = await glob.hashFiles(this.cacheDependencyPath);
-    const primaryKey = `${this.CACHE_KEY_PREFIX}-${process.env['RUNNER_OS']}-python-${this.pythonVersion}-${this.packageManager}-${hash}`;
-    const restoreKey = `${this.CACHE_KEY_PREFIX}-${process.env['RUNNER_OS']}-python-${this.pythonVersion}-${this.packageManager}`;
+    let primaryKey = '';
+    let restoreKey = '';
+
+    if (IS_LINUX) {
+      console.log('here');
```
Comment (dmitry-shibanov) [inline src/cache-distributions/pip-cache.ts:64]: remove this log.

File: src/cache-distributions/pip-cache.ts [hunk 2]
```diff
@@ -57,8 +57,19 @@ class PipCache extends CacheDistributor {
...
+      const osRelease = await getLinuxOSReleaseInfo();
+      primaryKey = `${this.CACHE_KEY_PREFIX}-${process.env['RUNNER_OS']}-${osRelease}-python-${this.pythonVersion}-${this.packageManager}-${hash}`;
+      restoreKey = `${this.CACHE_KEY_PREFIX}-${process.env['RUNNER_OS']}-${osRelease}-python-${this.pythonVersion}-${this.packageManager}`;
+    } else {
+      console.log('here2');
```
Comment (dmitry-shibanov) [inline src/cache-distributions/pip-cache.ts:69]: remove this log

File: src/utils.ts [hunk 3]
```diff
@@ -119,3 +120,20 @@ export function isCacheFeatureAvailable(): boolean {
 
   return true;
 }
+
+export async function getLinuxOSReleaseInfo() {
+  const versionId = await exec.getExecOutput('lsb_release', ['-a'], {
+    silent: true
+  });
```
Comment (dmitry-shibanov) [inline src/utils.ts:127]:
```
const {stdout, stderr, exitCode} = await exec.getExecOutput('lsb_release', ['-a'], {
    silent: true
  });
```

File: src/utils.ts [hunk 4]
```diff
@@ -119,3 +120,20 @@ export function isCacheFeatureAvailable(): boolean {
...
+  let osVersion = '';
+  let osRelease = '';
+
+  versionId.stdout.split('\n').forEach(elem => {
+    if (elem.includes('Distributor')) osVersion = elem.split(':')[1].trim();
+    if (elem.includes('Release')) osRelease = elem.split(':')[1].trim();
+  });
+
+  core.info(osRelease);
+  core.info(osVersion);
```
Comment (dmitry-shibanov) [inline src/utils.ts:137]: I think we should convert it debug and change it to
```
core.debug(`OS Release: ${osRelease}, Version: ${osVersion}`)
```

File: src/utils.ts [hunk 5]
```diff
@@ -119,3 +120,20 @@ export function isCacheFeatureAvailable(): boolean {
 
   return true;
 }
+
+export async function getLinuxOSReleaseInfo() {
+  const versionId = await exec.getExecOutput('lsb_release', ['-a'], {
```
Comment (dmitry-shibanov) [inline src/utils.ts:125]: change command to `lsb_release -i -r -s`

File: src/utils.ts [hunk 6]
```diff
@@ -119,3 +120,20 @@ export function isCacheFeatureAvailable(): boolean {
...
+    silent: true
+  });
+  let osVersion = '';
+  let osRelease = '';
+
+  versionId.stdout.split('\n').forEach(elem => {
+    if (elem.includes('Distributor')) osVersion = elem.split(':')[1].trim();
+    if (elem.includes('Release')) osRelease = elem.split(':')[1].trim();
+  });
```
Comment (dmitry-shibanov) [inline src/utils.ts:134]: change it to 
```
const [osRelease, osVersion] = stdout.trim().split('\n');
```

**Review by vsafonkin — APPROVED** [review]

**Review by marko-zivic-93 — APPROVED** [review]

**Pull request merged**